streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNodeRecordDeserializer.java

```diff
@@ -54,4 +58,28 @@
                                     rawRecord.serializedValueSize(), key, value);
 
     }
+
+    public ConsumerRecord<Object, Object> tryDeserialize(final ProcessorContext processorContext,
+                                                         ConsumerRecord<byte[], byte[]> rawRecord) {
+
+        if (deserializationExceptionHandler == null) {
+            return deserialize(rawRecord);
+        }
+
+        // catch and process if we have a deserialization handler
+        try {
+            return deserialize(rawRecord);
+        } catch (Exception e) {
+            DeserializationExceptionHandler.DeserializationHandlerResponse response =
+                deserializationExceptionHandler.handle(processorContext, rawRecord, e);
+            if (response.id == DeserializationExceptionHandler.DeserializationHandlerResponse.FAIL.id) {
```

Comment:
Why not just `if(response == DeserializationExceptionHandler.DeserializationHandlerResponse.FAIL)` ?